playground/backend/cmd/server/http.go

```diff
@@ -19,12 +19,28 @@ import (
 	"beam.apache.org/playground/backend/internal/logger"
 	"context"
 	"net/http"
+	"os"
+	"path/filepath"
 )
 
 // listenHttp binds the http.Handler on the TCP network address
-func listenHttp(ctx context.Context, errChan chan error, envs environment.NetworkEnvs, handler http.Handler) {
-	logger.Infof("listening HTTP at %s\n", envs.Address())
-	if err := http.ListenAndServe(envs.Address(), handler); err != nil {
+func listenHttp(ctx context.Context, errChan chan error, envs *environment.Environment, handler http.Handler) {
+	address := envs.NetworkEnvs.Address()
+	logger.Infof("listening HTTP at %s\n", address)
+
+	mux := http.NewServeMux()
+	mux.Handle("/", handler)
+	mux.HandleFunc("/readiness", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could you please extract the function to logical block?

Reply:
Done.